drivers/network/netio/netio.c

```diff
@@ -0,0 +1,904 @@
+/*
+ * PROJECT:     NetIO driver
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     A more convenient networking (TCP/IP, UDP/IP) kernel API (NT6+)
+ * COPYRIGHT:   Copyright 2023-2024 Johannes Thoma <johannes@johannesthoma.com>
+ */
+
+/* We are a NT6+ driver .. */
+#undef _WIN32_WINNT
+#define _WIN32_WINNT 0x600
+#undef WINVER
+#define WINVER 0x600
```

Comment:
Just a question for understanding: I guess it doesn't compile without any of such hacky-undef'ing? What would fail without?

Reply:
Yes we need to tell the headers that we are NT6 else we get lots of compilation errors starting with:

    In file included from ../sdk/include/psdk/netiodef.h:10,
                 from ../sdk/include/ddk/netioddk.h:26,
                 from ../sdk/include/ddk/wsk.h:29,
                 from ../drivers/network/netio/netio.c:33:
    ../sdk/include/psdk/ws2def.h:364:3: error: unknown type name 'u_short'
       u_short sa_family;
       ^~~~~~~
    ../sdk/include/psdk/ws2def.h:530:3: error: unknown type name 'DWORD'
       DWORD dwBufferCount;
       ^~~~~
    ../sdk/include/psdk/ws2def.h:536:3: error: unknown type name 'DWORD'
       DWORD dwFlags;

and so on...

Reply:
Resolved, but I want to keep that discussion-thread uncollapsed for others to see. Doesn't block anything.

Reply:
Ok thanks for letting me know

Reply:
Hi I found that if I apply some minor patches to the headers it compiles without forcing the NT version to 6. Will push a commit in a minute...